wire: Add tests for MsgGetCFilter.

jozn opened the pull request:
This adds standard wire tests for MsgGetCFilter. Things done in these
commit:
- Add standard wire test for MsgGetCFilter
- Add an extended error checking for checking malformed encoded
    data return correct errors. For this Message type what has been done is similar to what WireErrors is doing but it has been put
    in a different test to be explicit and for documentation purpose.
- Removal of irrelevant doc and double space at msggetcfilter.go.

jozn commented:
Double space removal has been fixed.
If requested can delete `ExtendedErrors` too.

davecgh commented on wire/msggetcfilter_test.go:
```diff
@@ -0,0 +1,293 @@
+	hashStr := "000000000000007a59f30586c1003752956a8b55e6f741fd5f24c800cd5e5e8c"
+	blockHash, err := chainhash.NewHashFromStr(hashStr)
+	if err != nil {
+		t.Errorf("NewHashFromStr: %v", err)
```
This should probably be `t.Fatalf` because otherwise the test will keep running and it'll end up crashing due to a nil `blockHash`.

The same with the rest.  Typically, you only want to use `t.Errorf` when the tests are in a loop and you intentionally want to continue to the next one after reporting the error.

jozn commented:
Updated as requested.
Some other changes: 
- Add a semicolon to block 200,000 hash for readability. The reason block 200,000 has been chosen was that in test codes block 100,000 was bitcoin hash so I choose this number to not make any confusion.
- Changed `TestGetCFilteExtendedErrors` to `TestGetCFilterMalformedErrors` to make the functionality of this kind of tests more representative. Will change the other PR which used this name style.
